app/models/registration.rb

```diff
@@ -420,6 +420,11 @@ def series_registration_info
                                    .join(" + ")
   end
 
+  def can_be_waitlisted!
```

Comment:
Not too wild about the `can_be_` and `!` combination. I know that the usage of `!` is a bit overloaded between "throws an error" and "mutates in-place", but perhaps something like `check_waitlist!`-ish can do?

Reply:
Gone with `ensure_waitlist_eligibility!` which feels like it avoids sounding like a boolean check while keeping the method purpose clear